src/Engine/UnitActionType.java

```diff
@@ -469,4 +470,81 @@ public String name()
       return "DELETE";
     }
   }
+
+  public static class UnitProduce extends UnitActionType
+  {
+    private static final long serialVersionUID = 1L;
+    final UnitModel typeToBuild;
+    public UnitProduce(UnitModel type)
+    {
+      typeToBuild = type;
+    }
+
+    @Override
+    public GameActionSet getPossibleActions(GameMap map, Path movePath, Unit actor, boolean ignoreResident)
+    {
+      XYCoord moveLocation = new XYCoord(movePath.getEnd().x, movePath.getEnd().y);
```

Comment:
We really need a `Path.getEndCoord` convenience function.

Reply:
Yeeeees.